### DISCUSSION TITLE
Role-based access control for step execution

### DISCUSSION THREAD

**Discussion by updogliu** [argoproj/argo-workflows#8977] [Ideas]
We are investigating using Argo Workflows for our CI/CD pipeline.

There is a step that labeling an image as "Prod-ready".  Only the tech lead of the project is allowed to manually approve this step. So the workflow should suspend at this step, and wait for the tech lead to login and approve it.

Is there a way to express this kind of permission with Argo Workflows?

**Comment by DiegoFlop** [comment]
You can use https://argoproj.github.io/argo-workflows/intermediate-inputs/
but I believe that this feature isn't create for that purpose

**Comment by updogliu** [comment]
Thanks for the info. Is it possible to specify which users / roles are allowed to approve it?
